### DISCUSSION TITLE
How to change BLE MIDI Device's name?

### DISCUSSION THREAD

**Discussion by jhsa** [tttapa/Control-Surface#697] [General]
Hi, I am trying to change the Bluetooth MIDI device's name but I cannot find how to.

Some sketch I have found on the net says:

```
// To change the BLE device name modify the file:
// .\Arduino\libraries\Control_Surface\src\MIDI_Interfaces\BLEMIDI.hpp
// ( at line 22-> constexpr const char *BLE_MIDI_NAME = "Control Surface (BLE)"; )
```

but I cannot find this file anywhere..
thanks in advance for you help.

João

**Comment by tttapa** [comment]
It's here: 

https://github.com/tttapa/Control-Surface/blob/e6542665da8997fdf33aee5ff1e0b7383c7ebe00/src/MIDI_Interfaces/BLEMIDI/ESP32/midi-init.c#L53

**Reply by jhsa** [reply]
Thanks for your quick answer. 
And what about something like this, instead of having to modify the library? 
`MIDI_BLE.begin("My Midi Device");`

**Reply by tttapa** [reply]
I've just pushed an update, you can now add `MIDI_BLE.setName("My Midi Device");` to your setup (before `Control_Surface.begin()`).

**Reply by jhsa** [reply]
Thank you. will try it right now.. Will report.

**Reply by jhsa** [reply]
It seems to work, Thank You. GREAT ADDITION.
